### PR TITLE
Add scons 4.3.0

### PR DESCRIPTION
**Pull request by PucklaMotzer09**

### CODE REVIEW

**Comment by PucklaMotzer09** [comment]
```
/github/home/.xmake/packages/s/scons/4.3.0/5ea24eadb114429e93e924cc96d0cade/lib/python3.9/site-packages/ does NOT support .pth files
bad install directory or PYTHONPATH
```

I think `/github/home/.xmake/packages/s/scons/4.3.0/5ea24eadb114429e93e924cc96d0cade/lib/python3.9/site-packages/` should be added to `PYTHONPATH` according to [this](https://thelinuxcluster.com/2020/06/23/test-failed-does-not-support-pth-files/)

File: packages/s/scons/xmake.lua [hunk 1]
```diff
@@ -26,7 +27,7 @@ package("scons")
         io.writefile("build/doc/man/scons-time.1", "")
         io.writefile("build/doc/man/sconsign.1", "")
 
-        os.vrunv("python", {"setup.py", "install", "--prefix", package:installdir()})
+        os.vrunv("PYTHONPATH=\"" .. package:getenv("PYTHONPATH") .. "\" python", {"setup.py", "install", "--prefix", package:installdir()})
```
Comment (waruqi) [inline packages/s/scons/xmake.lua:30]: add_deps("python") has added all python envs to the current on_install

Comment (waruqi) [inline packages/s/scons/xmake.lua:30]: os.vrunv("xxx", {}, {envs = {PYTHONPATH = ""}})

**Comment by xq114** [comment]
https://github.com/xmake-io/xmake-repo/blob/master/packages/m/meson/xmake.lua#L22

```
{PYTHONPATH = package:installdir()}
```

**Comment by waruqi** [comment]
can't copy 'scons.1': doesn't exist or not a regular file

**Pull request merged**